tests/helpers_tests/test_condition.py

```diff
@@ -108,6 +112,206 @@ def test_evaluate__returns_null__when_condition_has_an_invalid_type_property(sel
+        user_versions = ['2.9']
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertTrue(result, custom_err_msg)
+
+    def test_semver_gt__returns_false(self):
+        semver_greater_than_2_0_condition_list = [['Android', "2.0", 'custom_attribute', 'semver_gt']]
+        user_versions = ['2.0.0', '1.9']
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertFalse(result, custom_err_msg)
+
+    def test_evaluate__returns_None__when_user_version_is_not_string(self):
+        semver_greater_than_2_0_condition_list = [['Android', "2.0", 'custom_attribute', 'semver_gt']]
+        user_versions = [True, 37]
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertIsNone(result, custom_err_msg)
+
+    def test_evaluate__returns_None__when_user_version_with_invalid_semantic(self):
+        semver_greater_than_2_0_condition_list = [['Android', "2.0", 'custom_attribute', 'semver_gt']]
+        user_versions = ['3.7.2.2', '+']
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertIsNone(result, custom_err_msg)
+
+    def test_compare_user_version_with_target_version_equal_to_0(self):
+        semver_greater_than_2_0_condition_list = [['Android', "2.0", 'custom_attribute', 'semver_gt']]
+        versions = [
+            ('2.0.1', '2.0.1'),
+            ('2.0.0', '2.0.0'),
+            ('2.9', '2.9.1'),
+            ('2.9.9', '2.9.9'),
+            ('2.9.9-beta', '2.9.9-beta'),
+            ('2.1', '2.1.0'),
+            ('2.1', '2.1.215'),
+            ('2', '2.12'),
+            ('2', '2.785.13')
```

Comment:
How is the evaluation equal to 0 here?

Reply:
responded above. only compare till target digits.